spec/periodicChecksSpec.js

```diff
@@ -0,0 +1,561 @@
+              state: 'changes_requested',
+            }
+          ],
+        }),
+      };
+
+      await robot.receive(payloadData);
+    });
+
+    it('should call periodic check module', () => {
+      expect(
+        periodicCheckModule.ensurePullRequestIsAssigned
+      ).toHaveBeenCalled();
+    });
+
+    it('should ping pr author', () => {
+      expect(github.issues.addAssignees).toHaveBeenCalled();
+      expect(github.issues.addAssignees).toHaveBeenCalledWith({
+        issue_number: 3,
+        assignees: ['author3'],
+        owner: 'oppia',
+        repo: 'oppia',
+      });
+    });
+
+    it('should assign pr author', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+      expect(github.issues.createComment).toHaveBeenCalledWith({
+        issue_number: 3,
+        owner: 'oppia',
+        repo: 'oppia',
+        body:
+          'Hi @author3, it looks like some changes were requested on this ' +
+          'pull request by @reviewer1. PTAL. Thanks!',
+      });
+    });
+  });
+
+  describe('when pull request has been approved and author has merging rights', () => {
+    beforeEach(async () => {
+      const approvedPR = pullRequests[3];
+      github.pulls = {
+        get: jasmine.createSpy('get').and.callFake((params) => {
+          return {
+            data: pullRequests[params.pull_number - 1],
+          };
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [approvedPR, assignedPullRequest],
+        }),
+      };
+
+      await robot.receive(payloadData);
+    });
+
+    it('should call periodic check module', () => {
+      expect(
+        periodicCheckModule.ensurePullRequestIsAssigned
+      ).toHaveBeenCalled();
+    });
+
+    it('should check if pr author has merging rights', () => {
+      expect(github.orgs.checkMembership).toHaveBeenCalled();
+      expect(github.orgs.checkMembership).toHaveBeenCalledWith({
+        org: 'oppia',
+        username: 'author4',
+      });
+    });
+
+    it('should ping pr author', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+      expect(github.issues.createComment).toHaveBeenCalledWith({
+        issue_number: 4,
+        owner: 'oppia',
+        repo: 'oppia',
+        body:
+          'Hi @author4, this PR is ready to be merged. Please address any ' +
+          'remaining comments prior to merging, and feel free to merge ' +
+          "this PR once you're happy with it. Thanks!",
+      });
+    });
+
+    it('should assign pr author', () => {
+      expect(github.issues.addAssignees).toHaveBeenCalled();
+      expect(github.issues.addAssignees).toHaveBeenCalledWith({
+        issue_number: 4,
+        owner: 'oppia',
+        repo: 'oppia',
+        assignees: ['author4'],
+      });
+    });
+  });
+
+  describe('when pull request has been approved and has a changelog label but author does not have merging rights', () => {
```

Comment:
Break line - please check the complete file.

Reply:
Done.